Editor/Utility/CinemachineEditorAnalytics.cs

```diff
@@ -0,0 +1,260 @@
+            EditorAnalytics.SendEventWithLimit("cm_create_vcam", data);
+        }
+
+        struct CreateEventData
+        {
+            public string vcam_created; // vcam created from Create -> Cinemachine menu
+        }
+
+        /// <summary>
+        /// Send analytics event when using entering playmode
+        /// </summary>
+        /// <param name="state">State change to detect entering playmode</param>
+        static void SendAnalyticsOnPlayEnter(PlayModeStateChange state)
+        {
+            // Only send analytics if it is enabled
+            if (!EditorAnalytics.enabled)
+                return;
+
+            // Only send data when entering playmode
+            if (state != PlayModeStateChange.EnteredPlayMode)
+                return;
+
+            var startTime = Time.realtimeSinceStartup;
+
+            var cinemachineCore = CinemachineCore.Instance;
+            var vcamCount = cinemachineCore.VirtualCameraCount;
+            var vcamDatas = new List<VcamData>();
+
+            // collect data from all vcams
+            for (int i = 0; i < vcamCount; ++i)
+            {
+                var vcamBase = cinemachineCore.GetVirtualCamera(i);
+                CollectData(vcamBase, i.ToString(), ref vcamDatas);
+            }
+
+            var projectData = new ProjectData
+            {
+                brain_count = cinemachineCore.BrainCount,
+                vcam_count = cinemachineCore.VirtualCameraCount,
+                cam_count = Camera.allCamerasCount,
+                vcams = vcamDatas,
+                time_elapsed = Time.realtimeSinceStartup - startTime,
+            };
+
+            // Register our event
+            EditorAnalytics.RegisterEventWithLimit("cm_vcams_on_play", 
+                k_MaxEventsPerHour, k_MaxNumberOfElements, k_VendorKey);
+            
+            // Send the data to the database
+            EditorAnalytics.SendEventWithLimit("cm_vcams_on_play", projectData);
+        }
+
+        static void CollectData(CinemachineVirtualCameraBase vcamBase, string id, ref List<VcamData> vcamDatas)
+        {
+            if (vcamBase == null) return;
+            
+            var vcamData = new VcamData(id, vcamBase);
+            GetExtensions(vcamBase, out vcamData.extensions, out vcamData.custom_extension_count);
+            
+            // VirtualCamera
+            var vcam = vcamBase as CinemachineVirtualCamera;
+            if (vcam != null)
+            {
+                vcamData.blend_hint = vcam.m_Transitions.m_BlendHint.ToString();
+                vcamData.inherit_position = vcam.m_Transitions.m_InheritPosition;
+                vcamData.mode_overwrite = vcam.m_Lens.ModeOverride.ToString();
+
+                // collect components on vcam
+                GetComponents(vcam.GetComponentPipeline(), out vcamData.body_component, out vcamData.aim_component, 
+                    out vcamData.noise_component, out vcamData.custom_component_count);
```

Comment:
This is nearly repeated for new vcam. But, I felt it is cleaner this way.

Reply:
It's ok.  This file in general is looking much better and more readable.

Reply:
It can be improved still further by moving the methods GetExtensions(), GetComponents(), and GetTypeName() directly into VcamData as non-static methods - many parameters that you are now passing will just vanish and the calls will be very pretty and not painful to repeat.